x/profile/types/genesis_test.go

```diff
@@ -6,9 +6,140 @@ import (
+		consAddr4 = sample.AccAddress()
+		addr1     = sample.AccAddress()
+		addr2     = sample.AccAddress()
+		addr3     = sample.AccAddress()
+		addr4     = sample.AccAddress()
+	)
+	tests := []struct {
+		name     string
+		genState *GenesisState
+		err      error
+	}{
+		{
+			name:     "default is valid",
+			genState: DefaultGenesis(),
+		}, {
+			name: "valid custom genesis",
+			genState: &GenesisState{
+				ConsensusKeyNonceList: []*ConsensusKeyNonce{
+					{ConsAddress: consAddr1, Nonce: 1},
+					{ConsAddress: consAddr2, Nonce: 2},
+					{ConsAddress: consAddr3, Nonce: 3},
+					{ConsAddress: consAddr4, Nonce: 4},
+				},
+				ValidatorByAddressList: []*ValidatorByAddress{
+					{Address: addr1, ConsensusAddress: consAddr1},
+					{Address: addr2, ConsensusAddress: consAddr2},
+					{Address: addr3, ConsensusAddress: consAddr3},
+					{Address: addr4, ConsensusAddress: consAddr4},
+				},
+				ValidatorByConsAddressList: []*ValidatorByConsAddress{
+					{ConsAddress: consAddr1, Address: addr1},
+					{ConsAddress: consAddr2, Address: addr2},
+					{ConsAddress: consAddr3, Address: addr3},
+					{ConsAddress: consAddr4, Address: addr4},
+				},
+			},
+		}, {
+			name: "duplicated consensus key",
+			genState: &GenesisState{
+				ConsensusKeyNonceList: []*ConsensusKeyNonce{
+					{ConsAddress: consAddr1, Nonce: 1},
+					{ConsAddress: consAddr1, Nonce: 1},
+				},
+			},
+			err: fmt.Errorf("duplicated index for consensusKeyNonce: %s", consAddr1),
+		}, {
+			name: "duplicated validator by address",
+			genState: &GenesisState{
+				ValidatorByAddressList: []*ValidatorByAddress{
+					{Address: addr1, ConsensusAddress: consAddr1},
+					{Address: addr1, ConsensusAddress: consAddr1},
+				},
+			},
+			err: fmt.Errorf("duplicated index for validatorByAddress: %s", addr1),
+		}, {
+			name: "consensus key not found",
+			genState: &GenesisState{
+				ConsensusKeyNonceList: []*ConsensusKeyNonce{
+					{ConsAddress: consAddr1, Nonce: 1},
+					{ConsAddress: consAddr2, Nonce: 2},
+				},
+				ValidatorByAddressList: []*ValidatorByAddress{
+					{Address: addr1, ConsensusAddress: consAddr1},
+					{Address: addr1, ConsensusAddress: consAddr2},
+				},
+			},
+			err: fmt.Errorf("consesus key not found for address: %s", consAddr1),
+		}, {
+			name: "duplicated validator by consensus address",
+			genState: &GenesisState{
+				ConsensusKeyNonceList: []*ConsensusKeyNonce{
+					{ConsAddress: consAddr1, Nonce: 1},
+					{ConsAddress: consAddr2, Nonce: 2},
+				},
+				ValidatorByAddressList: []*ValidatorByAddress{
+					{Address: addr1, ConsensusAddress: consAddr1},
+					{Address: addr2, ConsensusAddress: consAddr2},
+				},
+				ValidatorByConsAddressList: []*ValidatorByConsAddress{
+					{ConsAddress: consAddr1, Address: addr1},
+					{ConsAddress: consAddr1, Address: addr1},
+				},
+			},
+			err: fmt.Errorf("duplicated index for validatorByConsAddress: %s", consAddr1),
+		}, {
+			name: "validator by consensus address not found",
```

Comment:
Can we add a case where providing a `ValidatorByAddress` with no `ConsensusAddress` is valid